gradoop-temporal/src/main/java/org/gradoop/temporal/model/impl/TemporalGraph.java

```diff
@@ -0,0 +1,229 @@
+    return this.config.getTemporalGraphFactory();
+  }
+
+  @Override
+  public BaseGraphCollectionFactory<TemporalGraphHead, TemporalVertex, TemporalEdge, TemporalGraph,
+      TemporalGraphCollection> getCollectionFactory() {
+    return this.config.getTemporalGraphCollectionFactory();
+  }
+
+  /**
+   * Returns a 1-element dataset containing a {@link Boolean} value which indicates if the graph is empty.
+   *
+   * A graph is considered empty, if it contains no vertices.
+   *
+   * @return  1-element dataset containing {@link Boolean#TRUE}, if the collection is
+   *          empty or {@link Boolean#FALSE} if not
+   */
+  public DataSet<Boolean> isEmpty() {
+    return getVertices().map(new True<>()).distinct()
+      .union(getConfig().getExecutionEnvironment().fromElements(false)).reduce(new Or())
+      .map(new Not());
+  }
+
+  /**
+   * Writes the graph to given data sink.
+   *
+   * @param dataSink The data sink to which the graph should be written.
+   * @throws IOException if the graph can't be written to the sink
+   */
+  public void writeTo(TemporalDataSink dataSink) throws IOException {
+    dataSink.write(this);
+  }
+
+  /**
+   * Writes the graph to given data sink with an optional overwrite option.
+   *
+   * @param dataSink The data sink to which the graph should be written.
+   * @param overWrite determines whether existing files are overwritten
+   * @throws IOException if the graph can't be written to the sink
+   */
+  public void writeTo(TemporalDataSink dataSink, boolean overWrite) throws IOException {
+    dataSink.write(this, overWrite);
+  }
+
+  @Override
+  public boolean isGVELayout() {
+    return this.layout.isGVELayout();
+  }
+
+  @Override
+  public boolean isIndexedGVELayout() {
+    return this.layout.isIndexedGVELayout();
+  }
+
+  @Override
+  public DataSet<TemporalGraphHead> getGraphHead() {
+    return this.layout.getGraphHead();
+  }
+
+  @Override
+  public DataSet<TemporalVertex> getVertices() {
+    return this.layout.getVertices();
+  }
+
+  @Override
+  public DataSet<TemporalVertex> getVerticesByLabel(String label) {
+    return this.layout.getVerticesByLabel(label);
+  }
+
+  @Override
+  public DataSet<TemporalEdge> getEdges() {
+    return this.layout.getEdges();
+  }
+
+  @Override
+  public DataSet<TemporalEdge> getEdgesByLabel(String label) {
+    return this.layout.getEdgesByLabel(label);
+  }
+
+  //----------------------------------------------------------------------------
+  // Unary Operators
+  //----------------------------------------------------------------------------
+
+  @Override
+  public TemporalGraph diff(TemporalPredicate firstSnapShot, TemporalPredicate secondSnapshot) {
+    return callForGraph(new Diff(firstSnapShot, secondSnapshot));
+  }
+
+  @Override
+  public TemporalGraphCollection query(String query, String constructionPattern, boolean attachData,
+    MatchStrategy vertexStrategy, MatchStrategy edgeStrategy, GraphStatistics graphStatistics) {
```

Comment:
Same here, this calls an operator, not just another method of the class (like the other `query` methods do). I think this should stay here (or be moved to a base class later, since this is generified).